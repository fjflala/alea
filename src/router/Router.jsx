/**
 * Module dependencies
 */
import React from 'react';
import {
  Switch,
  Route,
  BrowserRouter,
} from 'react-router-dom';

/**
 * UI Components
 */
import PrivateRoute from '../components/PrivateRoute';

/**
 * Pages
 */
import Login from '../pages/Login';
import Signup from '../pages/Signup';
import Users from '../pages/UsersList';
import EditUser from '../pages/EditUser';
import CreateUser from '../pages/CreateUser';
import NotFound from '../pages/NotFound';

/**
 * Represents a router of an application
 */
const Router = () => (
  <BrowserRouter>
    <Switch>
      <Route path="/" exact render={() => (<div>HOME</div>)} />
      <Route path="/login" component={Login} />
      <Route path="/signup" component={Signup} />
      <PrivateRoute path="/users/" exact component={Users} />
      <PrivateRoute path="/users/create" component={CreateUser} />
      <PrivateRoute path="/users/:userID/edit" component={EditUser} />
      <Route path="*" component={NotFound} />
    </Switch>
  </BrowserRouter>
);

/**
 * Expose component
 */
export default Router;
