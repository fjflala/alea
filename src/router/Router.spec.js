import React from 'react';
import renderer from 'react-test-renderer';
import Router from './Router';
import { Router as RouterDOM } from 'react-router-dom';
import { createMemoryHistory } from 'history';

describe('<Router />', () => {
  it('should correctly render', () => {
    const tree = renderer
      .create(<RouterDOM history={createMemoryHistory({route: ['/']})}>
        <Router />
      </RouterDOM>)
      .toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});