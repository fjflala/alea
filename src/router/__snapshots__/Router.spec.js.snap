// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Router /> should correctly render 1`] = `
<div>
  HOME
</div>
`;
